### PR TITLE
typings(ChannelLogsQueryOptions): update before and after options

### PR DESCRIPTION
**Pull request by Fyko**
**Please describe the changes this PR makes and why it should be merged:**

This PR updates the `ChannelLogsQueryOptions` interface changing the `before` and `after` types to both Snowflake and number. 

This is done to accommodate the ability to fetch messages based on their order in a channel (for example, the 10th message of the channel). 

Example code to return the first message of a channel:
```js
<TextChannel>.fetch({ limit: 1, after: 1 });
``` 

**Status**
- [ ] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by NotSugden** [comment]
perhaps update the jsdoc too in the MessageManager

**Comment by SpaceEEC** [comment]
This behavior does not seem to be [documented](https://discordapp.com/developers/docs/resources/channel#get-channel-messages), might be subject-to-change or even a bug on Discord's end.

**Comment by almostSouji** [comment]
> This is done to accommodate the ability to fetch messages based on their order in a channel (for example, the 10th message of the channel).

This is a misconception on your end.

The snowflake passed to the respective option before/after/around are resolved to their timestamp portion, providing 0, 1, 10, 100 etc. will all yield the same (and at that expected) result of the first `limit` message(s) in the channel.

I checked (bbdbc4cfa789383b7b3dbecf5e6b8401ea2dd998 ), and both `{ limit: 1, around: 1 }` and `{ limit: 2, around: 1 }` yield expected results of one message, which is the first in the channel (again, the same is the case for 10, 100, etc. further disproving the "on their order in a channel" statement.

> also, using a number for `after` seems to work fine, but `before` returns an empty collection

Now if we can assume above to be true it also makes sense why `{ before: x }` could not yield any messages, you are looking for the message(s) **before** the first message in the channel.

**TL;DR:**
If anything that should be Timestamp (or just leave it as snowflake as documented from the API)

**Comment by NotSugden** [comment]
> This behavior does not seem to be [documented](https://discordapp.com/developers/docs/resources/channel#get-channel-messages), might be subject-to-change or even a bug on Discord's end.

image

its also worth mentioning, as i believe souji pointed out,
> providing 0, 1, 10, 100 etc. will all yield the same (and at that expected) result
the API returns `limit` message(s) after the ID of whatever `after` is

**Comment by appellation** [comment]
`Snowflake` is the correct type here. Some argument could be made for including `number` in the Snowflake typedef, but I don't think that would be a good idea since most actual snowflakes are strings. If you want to use "fake" snowflakes as in your example, you can just make them into strings and get the expected result.
